charts/consul/values.yaml

```diff
@@ -2090,6 +2090,10 @@ syncCatalog:
   # global.enabled.
   enabled: false
 
+  # True if you want to deregister all services in this cluster from consul that have been registered by the catalog sync when the `enabled` flag is set to false.
+  # @type: boolean
+  purgeServicesOnDisable: false
```

Comment:
this needs a better name

Reply:
Also, as we discussed before, do we need to add more fields here to allow users to control the purging in more granular ways?

Reply:
we don't, it's not really a requirement and since we're doing this on a disable of the catalog-sync we don't want to potentially leave behind services that might no longer exist. The main requirement is "remove services registered by catalog sync when catalog sync is disabled"

Reply:
this still needs a better name

Reply:
Just brain dumping other possibilities:
- `cleanUpServicesOnDisable`
- `cleanUpOnDisable` (might be my vote thus far 🤷‍♂️ )
- `deregisterServicesOnDisable`
- `deregisterOnDisable`

It feels like the name needs to be some version of `..OnDisable` because putting `disable` as a prefix makes it seem like you are disabling something additional.

Reply:
yeah I don't like the prefix `disable` I was thinking maybe `cleanupNodeOnRemoval` since the `Disable` can be a bit of misnomer because this runs on uninstall of consul and when you disable `syncCatalog` whereas the `removal` can mean "removal of sync catalog" or "removal of consul"